server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -47,6 +47,7 @@
 
 public class RestRequest implements ToXContent.Params {
 
+    public static final String RESPONSE_RESTRICTED = "responseRestricted";
```

Comment:
I wonder whether this should be named more genericly instead of specific to response. I can image that a transport action class performs differently based on this parameter so it is not entirely response restricted.

Reply:
I am open to suggestions, but this the best name I could come up with. The known use case is to restrict the response and if turns out this name doesn't fit other cases we can change it. 

My rationale for this name - this is intended to be a "restriction" or limited version of the original thing and restriction/restricted is always relative to the thing being restricted.  Using just "restricted" raises the question, what exactly is restricted ? "response" is the thing we know we need to restrict and imo hinting how to formulate the "response" is responsibility of the RestRequest. I toyed around with "operatorRestricted" ..but imo "operator" doesn't really belong in a rest object and it really should be "nonOperatorRestricted" which imo is less intuitive in it's purpose.